SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -362,6 +372,76 @@ void ASpatialDebugger::OnToggleConfigUI()
 	}
 }
 
+void ASpatialDebugger::OnSelectActor()
+{
+	if (HitActors.Num() > 0)
+	{
+		TWeakObjectPtr<AActor> SelectedActor = HitActors[HoverIndex];
+
+		if (SelectedActor.IsValid())
+		{
+			if (SelectedActors.Contains(SelectedActor))
+			{
+				// Already selected so deselect
+				SelectedActors.Remove(SelectedActor);
+			}
+			else
+			{
+				// Add selected actor to enable drawing tags
+				SelectedActors.Add(SelectedActor);
+			}
+		}
+	}
+}
+
+void ASpatialDebugger::OnHighlightActor()
+{
+	HoverIndex++;
+	ValidateHoverIndex();
+}
+
+void ASpatialDebugger::ToggleSelectActor()
+{
+	bSelectActor = !bSelectActor;
+	if (bSelectActor)
+	{
+		if (CrosshairTexture != nullptr)
+		{
+			// Hide the mouse cursor as we will draw our own custom crosshair
+			LocalPlayerController->bShowMouseCursor = false;
```

Comment:
Lets cache what this was set to, and restore it, so we don't enable a cursor that was previously intentional hidden

Reply:
Sure, I've added a check to see if the config window is open here as this is the safest option.

Reply:
Looks good - is it possible to close the debugger whilst selecting actors?

Reply:
Yes, you can still select actors with the debugger closed but you need to reopen it to "Stop selecting actors" when you are finished.